src/ir/types/schema.h

```diff
@@ -22,6 +23,40 @@ class Schema {
 
   friend class TypeFactory;
 
+  //return a string representing the Schema in the following format: 
+  // name: <name or empty>
+  // fields: 
+  //    <field_name>: <field_kind>
+  //... 
+  //only showing the first layer of the schema, e.g. not returning the values of <field_kind>
+  //for easier testing, fields are sorted alphabetically based on their name  
+  std::string ToString(){
+    
+    std::string stringValue = absl::StrCat("name: ", name_.value_or(""), "\n", "fields: \n"); 
+
+    std::vector<std::string> fields_to_string; 
+  
+    for (auto& pair : fields_) {
```

Comment:
You can use destructuring in the loop directly:

```
for (auto &[name, kind] : fields_) {
...
}
```

This comment is moot if you use `absl::StrJoin`.